lib/rack/twilio_webhook_authentication.rb

```diff
@@ -17,18 +17,20 @@ module Rack
   # doesn't validate then the middleware responds immediately with a 403 status.
 
   class TwilioWebhookAuthentication
-    def initialize(app, auth_token, *paths)
+    def initialize(app, auth_token, *paths, &block)
       @app = app
       @auth_token = auth_token
+      define_singleton_method(:getAuthToken, block) if block_given?
       @path_regex = Regexp.union(paths)
     end
 
     def call(env)
       return @app.call(env) unless env["PATH_INFO"].match(@path_regex)
-      validator = Twilio::Util::RequestValidator.new(@auth_token)
       request = Rack::Request.new(env)
       original_url = request.url
       params = request.post? ? request.POST : {}
+      @auth_token ||=  getAuthToken(params['AccountSid'])
```

Comment:
Will this only execute `getAuthToken` once, as after the first run through `@auth_token` will be set. Might

```
auth_token = @auth_token || get_auth_token(params['AccountSid'])
validator = Twilio::Util::RequestValidator.new(auth_token)
```

be better?